### ISSUE TITLE
Trigger precaching on Service worker when User wants

### ISSUE THREAD

**Issue by crubier** [Labelflow/labelflow#399]
See https://github.com/Labelflow/labelflow/pull/397/files

**Comment by nicOwlas** [comment]
When the user clicks either on `skip tutorial` or `go to tutorial`?

**Comment by RaynalHugo** [comment]
I am not sure this would work exactly like that. 
According to workbox Documentation, precaching is the process of caching stuff before the SW finishes installing. 
https://developers.google.com/web/tools/workbox/guides/using-bundlers#optional_inject_a_precache_manifest

I am not sure we can simply call the `precacheAndRoute` function when we want after the SW has loaded.

**Comment by crubier** [comment]
I think you can actually. The PreCache function is just a simple function that downloads and put in the caches the files you give it as argument. Nothing magic.

**Comment by crubier** [comment]
The “magic” is more about generating that list of files (the cache manifest) which is done at build time by webpack and injected in the bundled code. But then at runtime it’s dead simple.

**Comment by RaynalHugo** [comment]
Yes but I think it does that synchronously and it's blocking the thread (that's why it deferred the end of the installation of the worker). I wonder if there isn't a smarter way to do this.

**Comment by crubier** [comment]
Precache things (network requests) is by definition asynchronous. So it is impossible that it is blocking the thread.

- if called during service worker install, `precacheAndRoute` probably calls `waitUntil` to delay the "WorkerReady" signal until it has finished precaching everything (asynchronously). (See https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle ). But it is not a blocking call. 
- it uses a lot of bandwidth because it downloads stuff, so it would slow down the rest

So I think calling it later on is fine. I'm not sure that "when the user clicks the start button" is the best time either. It's probably too soon, it will slow down important stuff at that point.
